src/preact.d.ts

```diff
@@ -277,8 +284,8 @@ declare namespace JSX {
 		charSet?:string;
 		challenge?:string;
 		checked?:boolean;
-		class?:string;
-		className?:string;
+		class?:string | Object;
+		className?:string | { [key:string]: string | boolean | number };
```

Comment:
now it occurred to me that you want an object for className, does that actually work?

maybe you wanted to use `style`?